include/bsp/iotc_test_echoserver.h

```diff
@@ -0,0 +1,81 @@
+#ifndef __IOTC_TEST_ECHOSERVER_H__
+#define __IOTC_TEST_ECHOSERVER_H__
+
+#include <netdb.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <unistd.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+const uint16_t kTimeoutSeconds = 1;
+const uint16_t kBufferSize = 1024;
+class EchoTestServer {
+public:
+  EchoTestServer(uint16_t sockType, uint16_t port)
+      : sock_type_(sockType), test_port_(port) {}
+
+  uint16_t CreateServer();
+  uint16_t RunServer();
+  void StopServer();
+  char* get_recv_buf();
+
+private:
+  uint16_t sock_type_, test_port_;
```

Comment:
Pay attention to declaration order:
https://google.github.io/styleguide/cppguide.html#Declaration_Order

Private function declarations should come first, then members.